pkg/operator/ceph/csi/util.go

```diff
@@ -104,15 +104,27 @@ func templateToDeployment(name, templatePath string, p templateParam) (*apps.Dep
 	return &ds, nil
 }
 
-func getToleration(provisioner bool) []corev1.Toleration {
+func getToleration(clientset kubernetes.Interface, namespace string, provisioner bool) []corev1.Toleration {
 	// Add toleration if any
 	tolerations := []corev1.Toleration{}
 	var err error
 	tolerationsRaw := ""
-	if provisioner {
-		tolerationsRaw = os.Getenv(provisionerTolerationsEnv)
-	} else {
-		tolerationsRaw = os.Getenv(pluginTolerationsEnv)
+	// TODO: export config map name
+	rookOperatorConfig, err := k8sutil.GetConfigMap(clientset, "rook-operator-config", namespace)
+	// if config map exists and has tolerations specified, disregard corresponding Env Var
+	if err == nil && rookOperatorConfig.Data != nil {
+		if provisioner {
+			tolerationsRaw = rookOperatorConfig.Data[provisionerTolerationsEnv]
+		} else {
+			tolerationsRaw = rookOperatorConfig.Data[pluginTolerationsEnv]
+		}
+	}
+	if err != nil || rookOperatorConfig.Data == nil || tolerationsRaw == "" {
```

Comment:
we might miss other error messages  also, we need to check NotFound error here
```go
if err == apierrors.NotFound {
}
```

Reply:
In case of errors, we are defaulting to Env Var, specially if it is `NotFound`.